src/table/utils/get-view-state.ts

```diff
@@ -2,18 +2,18 @@ import { findPaginationVisibleRows, findVirtualizedVisibleRows } from './find-vi
 import getVisibleHeight from './get-visible-height';
 import { getTotalPosition } from '../../handle-data';
 import renderAsPagination from '../../render-as-pagination';
-import { TableLayout, ViewService } from '../../types';
+import { TableLayout, TotalsPosition, ViewService } from '../../types';
 
 const getViewState = (layout: TableLayout, viewService: ViewService, rootElement: HTMLElement) => {
-    if (viewService.viewState && !viewService.viewState.isMultiPage) return viewService.viewState;
-  
-    if (renderAsPagination(layout, viewService)) {
-      const totalsPosition = getTotalPosition(layout, viewService);
+  if (viewService.viewState && !viewService.viewState.isMultiPage) return viewService.viewState;
+    const isPagination = renderAsPagination(layout, viewService);
+    const totalsPosition = (isPagination || viewService.viewState?.isMultiPage) ? getTotalPosition(layout, viewService) : undefined;
```

Comment:
Why can't you just always resolve the `totalsPosition` object?

Reply:
We don't always need totalsPosition.

Reply:
But there is no harm in getting every time, it is not an expensive calculation. That removes all the added logic, and the need to test it

Reply:
Again it is no harm but then I need to update a number of unit tests which checking if getTotalPosition is executed or not. I always try to avoid to update old unit tests as much as possible. But if you insist then getTotalPosition is always executed.

Reply:
Then I see why you went with that approach. But I think that is an approach that builds technical debt. I think spending some extra minutes updating the tests, to write clean, easy to understand code is worth it. Especially when it is such a small change in the tests